plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergV2.java

```diff
@@ -364,6 +388,8 @@ private void writeEqualityDeleteToNationTable(Table icebergTable)
                 .forTable(icebergTable)
                 .rowSchema(deleteRowSchema)
                 .createWriterFunc(GenericParquetWriter::buildWriter)
+                .withSpec(partitionSpec.orElse(PartitionSpec.unpartitioned()))
+                .withPartition(new DummyStructLike())
```

Comment:
I didn't realize equality deletes could be assigned a partition, but that cool. 

So we can actually do a bit better in our delete file filtering. If an equality delete has a partition, we can remove it.

Reply:
Bump, just use a `PartitionData` instead of the `DummyStructLike`